Document the inventories metadata payload (#9136)
DataDog/datadog-agent: pkg/metadata/inventories/README.md

# Inventory Payload

> This payload populates some of the field, for the agent, in the `inventories` product in DataDog.

Reviewer: ```suggestion
This package populates some of the agent-related fields in the `inventories` product in DataDog.
```

This is enabled by default but can be turned off using `inventories_enabled` config.

The payload is sent every 10min (see `inventories_max_interval` in the config) or whenever it's updated with at most 1
update every 5 minutes (see `inventories_min_interval`).

# Content

The package offers 2 method to add data to the payload: `SetAgentMetadata` and `SetCheckMetadata`. As the name suggests,

> checks use `SetAgentMetadata` and each metadata is linked to a check ID. Everything agent-related uses the other one.

Reviewer: ```suggestion
checks use `SetCheckMetadata` and each metadata is linked to a check ID. Everything agent-related uses the other one.
```

Any part of the agent can add metadata to the inventory payload.

## Check metadata

> `SetCheckMetadata` register data per check instance. Metadata can include the check version, the version of the

Reviewer: ```suggestion
`SetCheckMetadata` registers data per check instance. Metadata can include the check version, the version of the
```

monitored software, ... It depends on each check.

## Agent metadata

> `SetAgentMetadata` register data about the agent itself.

Reviewer: ```suggestion
`SetAgentMetadata` registers data about the agent itself.
```

# Format

> The payload is a JSON dict with the following field

Reviewer: ```suggestion
The payload is a JSON dict with the following fields
```

> - `hostname` - **string**: the hostname of the agent as show on the status page.

Reviewer: ```suggestion
- `hostname` - **string**: the hostname of the agent as shown on the status page.
```

- `timestamp` - **int**: the timestamp when the payload was created.

> - `check_metadata` - **dict of string to list**: dict keys the check name and value are a list of the metadata for each

Reviewer: ```suggestion
- `check_metadata` - **dict of string to list**: dictionary with check names as keys; values are a list of the metadata for each
```

  instance of that check.
  Each instance is composed of:
    - `last_updated` - **int**: timestamp of the last metadata update for this instance

>     - `config.hash` - **string**: the instance ID for this instance (as shown in the status page).

Reviewer: Is this key `config.hash` or is `hash` a subkey of a dictionary?

Author: It's `config.hash`, not sure what it's a dict with  `hash` and `provider`.

Reviewer: Got it.  Maybe I should have asked "is that a literal `.` character in the key" (answer: yes)

Author: yes it is !

    - `config.provider` - **string**: where the configuration came from for this instance (disk, docker labels, ...).

>     - Any other metadata register by the instance (instance version, version of the software monitored, ...).

Reviewer: ```suggestion
    - Any other metadata registered by the instance (instance version, version of the software monitored, ...).
```
